mobile/src/feeds/home-feeds-screen.tsx

```diff
@@ -1,28 +1,87 @@
 import React from 'react';
-import { FlatList, Text, View} from 'react-native';
+import { FlatList, Text, View, StyleSheet, ListRenderItemInfo, RefreshControl } from 'react-native';
 import { TopNavBar } from '../topNavBar/top-nav-bar';
 
-const navBar =  new TopNavBar('Feeds');
-const dataList = [
-{key: 'First Feed'},
-{key: 'Second Feed'},
-{key: 'Third Feed'},
-{key: 'Fourth Feed'},
-{key: 'Fifth Feed'},
-{key: 'Sixth Feed'},
-{key: 'Seventh Feed'},
-{key: 'Ten'}
-];
+import { Employee } from '../reducers/organization/employee.model';
+import { EmployeesStore } from '../reducers/organization/employees.reducer';
+import { Feed } from '../reducers/feeds/feed.model';
+import { connect } from 'react-redux';
+import { AppState } from '../reducers/app.reducer';
 
+import { FeedListItem } from './feed';
 
-export class HomeFeedsScreen extends React.Component {
+import { screenStyles as styles } from './styles';
+
+const navBar = new TopNavBar('Feeds');
+
+interface FeedsScreenProps {
+    feeds: Feed[];
+    employees: EmployeesStore;
+}
+
+interface FeedsScreenState {
+    employees?: Employee[];
+}
+
+const mapStateToProps = (state: AppState): FeedsScreenProps => ({
+    feeds: state.feeds,
+    employees: state.organization.employees
+});
+
+class HomeFeedsScreenImpl extends React.Component<FeedsScreenProps, FeedsScreenState> {
     public static navigationOptions = navBar.configurate();
 
+    constructor(props: FeedsScreenProps) {
+        super(props);
+        this.state = {};
+    }
+
+    public componentWillReceiveProps(nextProps: Readonly<FeedsScreenProps>) {
+        const { feeds } = nextProps;
+        const employees = feeds.map(f => nextProps.employees.employeesById.get(f.employeeId));
+
+        if (employees.length > 0) {
+            this.setState({
+                employees: employees
+            });
+        }
+    }
+
+    public shouldComponentUpdate() {
+        return true;
+    }
+
     public render() {
         return (
-            <View>
-                <FlatList data = {dataList} renderItem ={({item}) => <Text style={{padding: 10, fontSize: 18, height: 81, textAlign: 'center'}}>{item.key}</Text>} />
-            </View>
+            <FlatList
+                style={styles.view}
+                keyExtractor={this.keyExtractor}
+                ItemSeparatorComponent={this.itemSeparator}
+                data={this.props.feeds}
+                extraData={this.state.employees}
+                renderItem={this.renderItem}
+                ListHeaderComponent={this.headerComponent} />
         );
     }
-}
\ No newline at end of file
+
+    private keyExtractor(item: Feed) {
+        return item ? item.messageId : '';
+    }
+
+    private itemSeparator() {
+        return <View style={styles.separator}></View>;
+    }
+
+    private headerComponent() {
+        return <Text style={styles.viewHeaderText}>News feed</Text>;
```

Comment:
Do we really need this text?
![image](https://user-images.githubusercontent.com/16804094/35814828-1e52843c-0aa8-11e8-80d6-908d4dc9f3e3.png)

Reply:
You'd better ask a designer about that

Reply:
I think, he did it before we added StackNavigator's header